### ISSUE TITLE
Whitelisting for input validation

### ISSUE THREAD

**Issue by johnbillion** [WordPress-Coding-Standards/WordPress-Coding-Standards#406]
The following code will raise a "Detected usage of a non-validated input variable" error, despite liberal use of comments pleading phpcs to keep quiet.

What am I doing wrong?

``` php
public static function current_url() {
    return ( is_ssl() ? 'https://' : 'http://' ) . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI']; // WPCS: input var okay; validation okay; sanitization okay.
}
```

**Comment by johnbillion** [comment]
Using version 0.5.0, by the way.

**Comment by westonruter** [comment]
Validated means you need to check if they exist. So this should do the trick:

```php
public static function current_url() {
    $current_url = null;
    if ( isset( $_SERVER['HTTP_HOST'] ) && $_SERVER['REQUEST_URI'] ) {
        $current_url = is_ssl() ? 'https://' : 'http://';
        $current_url .= wp_unslash( $_SERVER['HTTP_HOST'] ); // WPCS: sanitization okay
        $current_url .= wp_unslash( $_SERVER['REQUEST_URI'] ); // WPCS: sanitization okay
    }
    return $current_url;
}
```

**Comment by johnbillion** [comment]
Is it not possible to skip this with the `validation okay` comment?

**Comment by westonruter** [comment]
I don't believe so, per what @JDGrimes has done. See also #383.

Well, you could suppress all validation warnings by excluding that sniff error in your ruleset.

**Comment by JDGrimes** [comment]
No, there has never been whitelisting for validation as far as I know. Whitelisting for sanitization was added in #294.

**Comment by JDGrimes** [comment]
Related: #393, #210.

**Comment by tivnet** [comment]
I got it working as follows:

    if ( isset( $_SERVER['HTTP_HOST'] ) ) { // Input var okay.
       echo esc_html( wp_unslash( $_SERVER['HTTP_HOST'] ) ); // Input var okay; sanitization okay.
    }

**Comment by jrfnl** [comment]
Is this still something which needs addressing or can this issue be closed ?

**Comment by johnbillion** [comment]
The code from @tivnet above does the trick 👍
